ui/hooks/useDisplayName.ts

```diff
@@ -27,6 +29,7 @@ export function useDisplayNames(
 
   const nameEntries = useNames(nameRequests);
   const firstPartyContractNames = useFirstPartyContractNames(nameRequests);
+  const nftCollectionsMetadata = useNftCollectionsMetadata(nameRequests);
```

Comment:
we could consider not doing this if this setting is off, although its unclear how the settings should intersect 
<img width="329" alt="Screenshot 2024-07-05 at 9 36 26 AM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/7249ee72-f999-4a34-b22b-365be32dd097">

Reply:
Good call @bergeron, in the past we override some of the preferences like this. Since precise simulations are increasing the effectiveness of UIX we may want to ignore this.
I will clarify this with Barbara.

Reply:
We should continue to override this setting so we're able to display more readable and meaningful simulations for users. We do have a simulations settings within the privacy settings and within advanced onboarding so users can still disable this functionality if they desire. Thank you for calling this out, though @bergeron!